src/pages/bucketlists.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { Modal, Button } from 'react-bootstrap';
+
+import Bucketlist from '../components/bucketlist';
+import BucketlistStore from '../stores/BucketlistStore';
+import * as bucketlistactions from '../actions/bucketlistactions';
+
+export default class UserBucketlists extends React.Component{
+    constructor(){
+        super();
+        this.state = {
+            user_bucketlists: [],
+            name: "",
+            date: "",
+            description: "",
+            show_modal: false,
+            editid: null,
+        };
+    }
+    componentWillMount(){
+        BucketlistStore.retrieveBucketlists(localStorage.getItem("token"));
+        BucketlistStore.on('change', () => {
+            this.setState({
+                user_bucketlists: BucketlistStore.getAll()});
+        });
+    }
+    handleChange(event){
+        //Validate the date
+        this.setState({
+         [event.target.id]: event.target.value,
+        });
+    }
+    handleSubmit(e){
+        e.preventDefault();
+        const payload = {
+            "name" : this.state.name,
+            "date" : this.state.date,
+            "description": this.state.description
+        }
+        if(this.state.editid){
+            bucketlistactions.editBucketlist(this.state.editid, payload);
+            this.close();
+        }else{
+            //throw in the token and call for a create-bucketlist action
+            bucketlistactions.createBucketlist(payload);
+        }        
+    }
+    handleDelete(id){
+        bucketlistactions.deleteBucketlist(String(id));
+    }
+    handleView(id, name){
+        //route to the retrieved list view
+        var url = '/bucketlists/'+id+'/items';
+        localStorage.setItem('list_id', id);
+        localStorage.setItem('list_name', name);
```

Comment:
Apart from this what other information are you storing on local storage and is it really necessary to have it stored on local storage?